pilot/pkg/security/authz/converter/converter.go

```diff
@@ -395,24 +386,31 @@ func (ug *Converter) v1alpha1ModelTov1beta1Policy(v1alpha1Model *authz_model.Mod
 		}
 	}
 
-	if len(accessRule.Services) == 0 {
-		authzConfig := createAuthzConfig("all-workloads", nil)
-		ug.AuthorizationPolicies = append(ug.AuthorizationPolicies, authzConfig)
-	}
-	for _, service := range accessRule.Services {
-		for j, selector := range ug.getSelectors(service, namespace) {
-			name := fmt.Sprintf("service-%s-%d", strings.ReplaceAll(service, "*", "wildcard"), j)
-			authzConfig := createAuthzConfig(name, &v1beta1.WorkloadSelector{
-				MatchLabels: selector,
-			})
-			ug.AuthorizationPolicies = append(ug.AuthorizationPolicies, authzConfig)
+	for _, accessRule := range v1alpha1Model.Permissions {
+		operation, err := convertAccessRuleToOperation(&accessRule)
+		if err != nil {
+			return fmt.Errorf("cannot convert access rule to operation: %v", err)
+		}
+
+		if len(accessRule.Services) == 0 {
+			authzConfig := createAuthzConfig("all-workloads", nil, operation)
+			c.AuthorizationPolicies = append(c.AuthorizationPolicies, authzConfig)
+		}
+		for _, service := range accessRule.Services {
+			for j, selector := range c.getSelectors(service, namespace) {
+				name := fmt.Sprintf("service-%s-%d", strings.ReplaceAll(service, "*", "wildcard"), j)
```

Comment:
Nit: If we have a method to "sanitize" a K8s Name it would be good to use it here.  I am concerned about newlines and Unicode.  I realize you did not write this line, just moved it.